### PR TITLE
raftstore: add split unit test

### PR DESCRIPTION
**Pull request by BusyJay**
add split check unit test cases.

@siddontang @hhkbp2 PTAL

### CODE REVIEW

**Comment by hhkbp2** [comment]
LGTM

File: src/raftstore/store/worker/split_check.rs [hunk 1]
```diff
@@ -125,13 +127,13 @@ pub struct Task {
 }
 
 impl Task {
-    pub fn new(ps: &PeerStorage) -> Task {
+    pub fn new(engine: Arc<DB>, region: &Region) -> Task {
```
Comment (siddontang) [inline src/raftstore/store/worker/split_check.rs]: is it better to move engine to the Runner field directly now?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
CI failed @BusyJay

**Pull request merged**